tests/jballerina-unit-test/src/test/resources/test-src/query/query-expr-with-query-construct-type.bal

```diff
@@ -692,6 +692,307 @@ function testQueryConstructingTableWithOnConflictClauseHavingNonTableQueryInWher
+
+    Customer[] customerList1 = [c1, c2, c1];
+    Person[] personList = [p1, p2];
+
+    var customerMap1 = map from var customer in customerList1
+        join var person in personList
+        on customer.name equals person.lastName
+        select [customer.id.toString(), {
+            id: customer.id,
+            name: person.firstName,
+            age: person.age,
+            noOfItems: customer.noOfItems
+        }]
+        on conflict onConflictError;
+    assertEqual(customerMap1, onConflictError);
+
+    Customer[] customerList2 = [c1, c2];
+
+    var customerMap2 = map from var customer in customerList2
+        join var person in personList
+        on customer.name equals person.lastName
+        select [customer.id.toString(), {
+            id: customer.id,
+            name: person.firstName,
+            age: person.age,
+            noOfItems: customer.noOfItems
+        }]
+        on conflict onConflictError;
+    assertEqual(customerMap2, {"1":{"id":1,"name":"Amy","age":23,"noOfItems":12},"2":{"id":2,"name":"Frank","age":30,"noOfItems":5}});
+}
+
+function testMapConstructingQueryExprWithLimitClause() {
+    error onConflictError = error("Key Conflict", message = "cannot insert.");
+
+    Customer c1 = {id: 1, name: "Melina", noOfItems: 12};
+    Customer c2 = {id: 2, name: "James", noOfItems: 5};
+    Customer c3 = {id: 3, name: "Melina", noOfItems: 25};
+
+    Customer[] customerList = [c1, c2, c3];
+
+    map<Customer>|error customerMap1 = map from var customer in customerList.toStream()
+        where customer.name == "Melina"
+        limit 1
+        select [customer.name, {
+            id: customer.id,
+            name: customer.name,
+            noOfItems: customer.noOfItems
+        }]
+        on conflict onConflictError;
+    assertEqual(customerMap1, {"Melina":{"id":1,"name":"Melina","noOfItems":12}});
+}
+
+function testMapConstructingQueryExprWithOrderByClause() {
+    map<int>|error sorted1 = map from var e in [1, 2, 10, 3, 5, 20]
+                                order by e ascending
+                                select [e.toString(), e];
+    assertEqual(sorted1, {"1":1,"2":2,"3":3,"5":5,"10":10,"20":20});
+
+    map<int>|error sorted2 = map from var e in [1, 2, 10, 3, 5, 20]
+                                order by e descending
+                                select [e.toString(), e];
+    assertEqual(sorted2, {"20":20,"10":10,"5":5,"3":3,"2":2,"1":1});
+
+    var sorted3 = map from var e in ["1", "2", "10", "3", "5", "20"]
+                    order by e ascending
+                    select [e, e] on conflict ();
+    assertEqual(sorted3, {"1":"1","10":"10","2":"2","20":"20","3":"3","5":"5"});
+
+    var sorted4 = map from var e in [1, 2, 5, 4]
+                    let int f = e / 2
+                    order by f ascending
+                    select [f.toString(), e] on conflict error("Error");
+    assertEqual(sorted4, error("Error"));
+}
+
```

Comment:
Shall we also add some tests where the contextually expected type is a reference type?